source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c

```diff
@@ -320,22 +504,37 @@ BaseType_t xNetworkInterfaceOutput( NetworkBufferDescriptor_t * const pxBuffer,
          * the protocol checksum to have a value of zero. */
         pxPacket = ( ProtocolPacket_t * ) ( pxBuffer->pucEthernetBuffer );
 
+        #if ( ipconfigUSE_IPv6 != 0 )
+            ICMPPacket_IPv6_t * pxICMPPacket = ( ICMPPacket_IPv6_t * ) pxBuffer->pucEthernetBuffer;
+
+            if( ( pxPacket->xICMPPacket.xEthernetHeader.usFrameType == ipIPv6_FRAME_TYPE ) &&
+                ( pxICMPPacket->xIPHeader.ucNextHeader == ipPROTOCOL_ICMP_IPv6 ) )
+            {
+                /* The EMAC will calculate the checksum of the IP-header.
+                 * It can only calculate protocol checksums of UDP and TCP,
+                 * so for ICMP and other protocols it must be done manually. */
+                usGenerateProtocolChecksum( pxBuffer->pucEthernetBuffer, pxBuffer->xDataLength, pdTRUE );
+            }
+        #endif /* ipconfigUSE_IPv6 */
+
         if( ( pxPacket->xICMPPacket.xEthernetHeader.usFrameType == ipIPv4_FRAME_TYPE ) &&
```

Comment:
The driver should not bother about ipV4 frames, unless `ipconfigUSE_IPv4` is defined.

Reply:
Done